javascript-engine/src/index.ts

```diff
@@ -0,0 +1,99 @@
+import init, { Engine } from '@unleash/yggdrasil-wasm'
+import { checkResponse, readResponse } from './utils'
+import { Strategies, type Strategy } from './strategies'
+import type {
+  Context,
+  FeatureDefinition,
+  MetricsBucket,
+  State,
+  Variant
+} from './types'
+export * from './types'
+
+await init()
+
+export const DISABLED_VARIANT: Variant = {
+  name: 'disabled',
+  enabled: false,
+  featureEnabled: false
+}
+
+export class UnleashEngine {
+  private engine: Engine
+  private strategies: Strategies
+
+  constructor(strategies?: Strategy[]) {
+    this.engine = new Engine()
+
+    const knownStrategiesResponse = this.engine.builtInStrategies()
+    const knownStrategies = readResponse<string[]>(knownStrategiesResponse)
+
+    this.strategies = new Strategies(knownStrategies)
+
+    if (strategies != null) {
+      this.strategies.registerCustomStrategies(strategies)
+    }
+  }
+
+  public takeState(state: State): void {
+    const response = this.engine.takeState(state)
```

Comment:
I believe this beast will return you some warnings if the user sent it state that it didn't parse correctly. I think we should attempt to bubble that up or log it